test/suite/test_compact11.py

```diff
@@ -0,0 +1,190 @@
+        for k in range(start_key, num_keys):
+            c[k] = ('%07d' % k) + '_' + 'abcd' * ((value_size // 4) - 2)
+        c.close()
+
+    def turn_on_bg_compact(self, config = ''):
+        self.session.compact(None, f'background=true,{config}')
+        while not self.get_bg_compaction_running():
+            time.sleep(0.1)
+
+    def compare_bitmap(self, orig, new):
+        # Compare the bitmaps from the metadata. Once a bit is set, it should never
+        # be cleared. But new bits could be set. So the check is only: if the original
+        # bitmap has a bit set then the current bitmap must be set for that bit also.
+        #
+        # First convert both bitmaps to a binary string, accounting for any possible leading
+        # zeroes (that would be truncated off). Then compare bit by bit.
+        orig_bits = bin(int('1'+orig, 16))[3:]
+        new_bits = bin(int('1'+new, 16))[3:]
+        self.pr("Original bitmap in binary: " + orig_bits)
+        self.pr("Reopened bitmap in binary: " + new_bits)
+        for o_bit, n_bit in zip(orig_bits, new_bits):
+            if o_bit != '0':
+               self.assertTrue(n_bit != '0')
+
+    def parse_blkmods(self, uri):
+        meta_cursor = self.session.open_cursor('metadata:')
+        config = meta_cursor[uri]
+        meta_cursor.close()
+        # The search string looks like: ,blocks=feffff1f000000000000000000000000
+        # Obtain just the hex string.
+        b = re.search(',blocks=(\w+)', config)
+        self.assertTrue(b)
+        # The bitmap string after the = is in group 1.
+        blocks = b.group(1)
+        self.pr(f"parse_blkmods for '{uri}': {blocks}")
+        return blocks
+
+    def test_compact11(self):
+        # Create N tables.
+        uris = []
+        files = []
+        bitmaps = []
+        for i in range(self.num_tables):
+            uri = self.uri_prefix + f'_{i}'
+            uris.append(uri)
+            files.append(f'file:test_compact11_{i}.wt')
+            self.session.create(uri, self.create_params)
+
+        # Populate the first half of each table.
+        for uri in uris:
+            self.populate(uri, 0, self.table_numkv // 2, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Take a full backup that will be used for incremental backups later during the test.
+        os.mkdir(self.backup_incr)
+        self.initial_backup = True
+        self.take_full_backup(self.backup_incr)
+        self.initial_backup = False
+
+        # Insert the latter 50% in each table.
+        for uri in uris:
+            self.populate(uri, self.table_numkv // 2, self.table_numkv, self.value_size)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Delete 50% of each file.
+        for uri in uris:
+            self.delete_range(uri, 50 * self.table_numkv // 100)
+
+        # Write to disk.
+        self.session.checkpoint()
+
+        # Generate the bitmaps of each file.
```

Comment:
I suggest "Get the bitmaps ...". This code is only reading the bitmaps, not generating them.